crates/sui-indexer/src/apis/coin_api_v2.rs

```diff
@@ -124,8 +125,18 @@ impl CoinReadApiServer for CoinReadApiV2 {
             .map_err(Into::into)
     }
 
-    async fn get_total_supply(&self, _coin_type: String) -> RpcResult<Supply> {
-        unimplemented!()
+    async fn get_total_supply(&self, coin_type: String) -> RpcResult<Supply> {
+        let coin_struct = parse_to_struct_tag(&coin_type)?;
```

Comment:
you may want to call something like `Coin::is_coin` and  return early if it's not a coin

Reply:
This type is actually the type parameter of the coin, i.e. the `T` part of `Coin<T>` so `is_coin` won't work.